### PR TITLE
fix(atomic): render result list content when it's ready

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-2035

Easy fix, but hard to find! The result templates (default template included) did not have time to register before the render was trigger. There seem to already have a `ready` variable available.

### CODE REVIEW

**Comment by olamothe** [comment]
Good find !

File: packages/atomic/src/components/search/result-lists/result-list-common.tsx [hunk 1]
```diff
@@ -84,6 +84,7 @@ export interface RenderListOptions extends DisplayOptions {
   getContentOfResultTemplate(
     result: Result | FoldedResult
   ): HTMLElement | DocumentFragment;
+  ready: boolean;
```
Comment (btaillon) [inline packages/atomic/src/components/search/result-lists/result-list-common.tsx:87]: It looks like `ready` is set  in the `onReady` callback anyway, so perhaps we should handle it within `result-list-common`

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/search/result-lists/result-list-common.tsx (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/search/result-lists/result-list-common.tsx:87]: ready is a @State property in the components, state property change trigger re-renders, a private property in the list common wouldnt

File: packages/atomic/src/components/search/result-lists/result-list-common.tsx [hunk 2]
```diff
@@ -252,8 +253,12 @@ export class ResultListCommon {
     resultsPerPageState,
     setListWrapperRef,
     getContentOfResultTemplate,
+    ready,
   }: RenderListOptions) {
     this.updateBreakpoints?.(host);
+    if (!ready) {
+      return;
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/search/result-lists/result-list-common.tsx:260]: Hmm perhaps, but it's not noticeable, simply a race condition for the time it takes for the result template method registration to run. In the use case where its useful, the results are rendered after the search already has been executed.

**Pull request merged**